### PR TITLE
Applying intersectionobserver for faas.

### PR DESCRIPTION
**Pull request by seanchoi-dev**
**Test URLs:**
- Before: https://main--bacom--adobecom.hlx.page/?martech=off
- After: https://main--bacom--adobecom.hlx.page/?martech=off&milolibs=faas-intersectionobserver

- milo test URL: https://main--milo--adobecom.hlx.page/drafts/seanchoi/faas-link-test?martech=off

### CODE REVIEW

**Review by Brandon32 — APPROVED** [review]

**Comment by meganthecoder** [comment]
@seanchoi-dev I don't think this is a trivial change. Considering how many live faas forms we have on bacom, I'd prefer to have @Dli3 do some testing

File: libs/blocks/faas/faas.js [hunk 1]
```diff
@@ -10,4 +12,16 @@ export default async function init(a) {
...
   }
+};
+
+export default async function init(a) {
+  if (a.textContent.includes('no-lazy')) {
+    loadFaas(a);
+  } else {
+    createIntersectionObserver({
+      el: a,
+      options: { rootMargin: `${ROOT_MARGIN}px` },
```
Comment (chrischrischris) [inline libs/blocks/faas/faas.js:23]: super nit - may as well set `const ROOT_MARGIN = '1000px';`

**Review by chrischrischris — APPROVED** [review]

**Comment by meganthecoder** [comment]
@seanchoi-dev Also, it looks like the lighthouse check didn't run because you don't have a milo url

File: libs/blocks/faas/faas.js (hunk 1 above)
Comment (seanchoi-dev) [reply libs/blocks/faas/faas.js:23]: It is already set at line 4.

**Pull request merged**